data/lua/explode_ability.lua

```diff
@@ -0,0 +1,56 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+
+-- The explode event code
+wesnoth.game_events.add_repeating("die", function()
+    local ec = wesnoth.current.event_context
+
+    if not ec.x1 or not ec.y1 then
+        return
+    end
+
+    local u_exploder = wesnoth.units.get(ec.x1, ec.y1)
+
+    if not u_exploder or not u_exploder:matches { ability_type = "explode" } then
```

Comment:
I think this should be abulity_type_active,

Reply:
I think it _should_ be `ability_type_active` but actually _can't_ be because we have no way in Lua to open a weapon specials context.

We probably need something that returns a closable table so you can use it like this:

```lua
local ctx <close> = wesnoth.units.weapons_context(whatever.parameters)
if not u_exploder or not u_exploder:matches { ability_type_active = "explode" } then
```

I'm not sure if that's the best name for it, but hopefully you get the idea.

Reply:
I think the context should just be set during die events just like it's set during attack events.


But that wasn't my point, my point is that if we want to do that, the code below which reads damage type and damage amount should also ignore inactive abilities, and there is currently no way to do that.

Reply:
I think we need the context exposed to Lua regardless – even if it's not useful here, then it's useful in AI. But from what you're saying it seems like it should be useful here as well.